### ISSUE TITLE
Giving different variable names when destructuring structs

### ISSUE THREAD

**Issue by theogf** [JuliaLang/julia#52191]
Right now it's possible (and great) to do

```julia
struct Foo
 a
 b
end
(; a, b) = Foo(2, 3)
```

However, in some cases (we want to destructure multiple `Foo` in the same scope)  it would be great to be able to directly use different variable names and do something like

```julia
(; a1 = a, b1 = b) = Foo(2, 3)
(; a2 = a, b2 = b) = Foo(1, 2)
```

There is a similar pattern in Rust described [here](https://doc.rust-lang.org/book/ch18-03-pattern-syntax.html?highlight=destruct#destructuring-to-break-apart-values)

**Comment by StefanKarpinski** [comment]
It's not at all obvious to me whether the name of the field should go on the left or right side of the `=` in this construct, which makes me kind of skeptical of adding this. Pretty sure I would always forget. You were constructing a named tuple, you would write `(; a = a1, b = b1)` rather than `(; a1 = a, b1 = b)`... which made me curious if Rust really did it this way... and indeed, they do not. But that only reinforces my point about how unobvious the ordering is considering that you got it wrong in the motivating example.

**Comment by theogf** [comment]
That's a really good point. I was thiking as we assign `a1` and `b1` rather than the opposite, having them on the LHS would make more sense, one could see the RHS as the "scope" of the struct.

But the fact that there is this ambiguity definitely would make it complicated.

> how unobvious the ordering is considering that you got it wrong in the motivating example.

I was not trying to reproduce the exact Rust syntax, just wanted to show that they were doing something similar.

**Comment by Pangoraw** [comment]
This is also discussed in https://github.com/JuliaLang/julia/issues/51940.

As another example, [JavaScript has syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#syntax) for both setting a default value and assigning the name:

```javascript
const { a: a1 = aDefault, b = bDefault } = obj;
```

**Comment by stevengj** [comment]
Yes, as I commented in the other thread I would prefer to use `=` for default values, much like keyword arguments.

Analogous to JavaScript, if we want to have renaming we could use Julia's own `using` syntax of `as`, i.e.
```jl
(; a as a1 = aDefault, b = bDefault) = obj
```
